File: backend/internal/models/organization.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type OrgBranding struct {
	PrimaryColor *string `json:"primary_color,omitempty"`
	LogoDataURI  *string `json:"logo_data_uri,omitempty"`
	AppTitle     *string `json:"app_title,omitempty"`
}

type Organization struct {
	ID        uuid.UUID   `json:"id"`
	Name      string      `json:"name"`
	Slug      string      `json:"slug"`
	CreatedAt time.Time   `json:"created_at"`
	UpdatedAt time.Time   `json:"updated_at"`
	Branding  OrgBranding `json:"branding"`
}

type CreateOrganizationRequest struct {
	Name string `json:"name"`
	Slug string `json:"slug"`
}

type UpdateOrganizationRequest struct {
	Name *string `json:"name,omitempty"`
	Slug *string `json:"slug,omitempty"`
}

File: frontend/src/types/organization.ts
export type MembershipRole = 'admin' | 'editor' | 'viewer'

export interface OrgBranding {
  primary_color?: string | null
  logo_data_uri?: string | null
  app_title?: string | null
}

export interface Organization {
  id: string
  name: string
  slug: string
  created_at: string
  updated_at: string
  role?: MembershipRole
  branding?: OrgBranding
}

export interface UpdateBrandingRequest {
  primary_color?: string | null
  logo_data_uri?: string | null
  app_title?: string | null
}

export interface CreateOrganizationRequest {
  name: string
  slug: string
}

export interface UpdateOrganizationRequest {
  name?: string
  slug?: string
}

export interface Member {
  id: string
  user_id: string
  email: string
  name?: string
  role: MembershipRole
  created_at: string
}

export interface Invitation {
  token: string
  email: string
  role: MembershipRole
  expires_at: string
}

export interface CreateInvitationRequest {
  email: string
  role: MembershipRole
}

export interface UpdateMemberRoleRequest {
  role: MembershipRole
}
